epersonregistrations.md

```diff
@@ -0,0 +1,61 @@
+# EPerson registration Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**GET /api/eperson/registrations**
+
+As we don't have a use case to iterate over all the existent registrations, the main endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Single EPerson registration
+
+As we don't have a use case to retrieve an epersonregistration based on the email, the single endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Search EPerson registration
+**/api/eperson/registrations/search/findByToken?token=<:token>**
+
+Exposes the registered email address based on the token.  
+Also exposes whether it's a new user registration, or a password reset for an existing user.
+
+```json
+{
+  "email": "user@institution.edu",
+  "user": null,
```

Comment:
we can expose the expires date.

In the current version if an user starts the registration process and loss the token there is no way to "resend" the token again or extend the validity of a token so the sysadmin needs to manipulate the db directly.  We can add a parameter to the creation of the registration to force the resend of a token that would mean, delete the previous one and create a new one. For that it will be useful for angular to know that the token is expired so that it cannot be used to create a new account (before to try to do that) and offer to the user the opportunity to get a fresh token

Reply:
Expired tokens don't exist anymore (not sure why, but it has been removed from Java a long time ago). Tim also requested details about expired tokens in https://github.com/DSpace/Rest7Contract/pull/117#discussion_r400502098 but I'd prefer not to add too much code for expired tokens if there's no functionality for this in the backend

Reply:
I agree, since expiration tokens don't actually work right now (even though a DB column does exist for them), we can leave them out for now until they are reimplemented.  So, that'd be a separate task (which we may need to do for GDPR eventually, but need not be done by this PR or its implementation).